### DISCUSSION TITLE
Possible to have CUDA (and ML) running both in Win10 host and WSL2 Ubuntu?

### DISCUSSION THREAD

**Discussion by patfla** [microsoft/WSL#6589] [General]
CUDA wasn't working inside WSL2 Ubuntu until yesterday when I switched my Win10 host over to Insider Preview - now it works.  However, for various reasons, I go back and forth between wanting to run CUDA for ML in both host and WSL2.  Going then to the host I needed to install CUDA (also reset Windows yesterday).  This put on Nvidia driver 461.09 while the kit I downloaded to get CUDA working inside WSL2 was 465.62.  Now CUDA (per /usr/local/cuda/samples ... deviceQuery) no longer works inside WSL2.

I can reinstall 465.62 and I'd hope that CUDA works again inside WSL2 but I'd guess that CUDA will then no longer work in the Win10 host.

**Comment by WSLUser** [comment]
That is a NVIDIA bug and should be filed on their bug tracker. Newer versions of a driver shouldn't break using CUDA on Win10 itself. I can see resource issues existing if you attempt to run on both Win10 and WSL2 unless you have multi-GPU and specify only one GPU device per CUDA instance. To prevent crashes, NVIDIA likely blocked this use case. It can work if it's done as specified above but you'd need to be the one to configure that. NVIDIA simply needs to ensure their driver will support it.
